### PR TITLE
Feature/cli

### PR DESCRIPTION
**Pull request by iruzevic**
### Changed
- Complete refactor of the block registration process for faster and more efficient block registration.
- Custom post type and taxonomy registration labels usage is now more consistent and simplified.
- All enqueue methods now supports new scripts args introduced in WP 6.3.
- All enqueue methods now implements `manifestCache`interface for fetching manifest data.
- All AbstractGeolocation method now implements `manifestCache`interface for fetching manifest data.

### Added
- Internal caching in WordPress transients for faster manifest parsing and block registration.
- Blocks render method now uses the Helpers::render method for better performance and code readability.
- New ManifestCache CLI command with example.
- New CLI commands for config theme and plugin.
- - `Exception/InvalidBlock.php` method: `missingItemException`.
- `src/Exception/InvalidManifest.php` methods: `missingManifestKeyException`, `emptyOrErrorManifestException`, `notAllowedManifestPathException`, `notAllowedManifestPathItemException`, `missingCacheTopItemException`, `missingCacheSubItemException`.
- `src/Exception/InvalidPath.php` method: `missingDirectoryException`, `missingFileException`, `missingFileWithExampleException`, `wrongOrNotAllowedParentPathException`.
- New `src/Helpers/ApiTrait.php` helper for better API handling.
- New `recursiveArrayFind` helper for finding items in multidimensional arrays recursively.
- New casing helpers `camelToSnakeCase`, `kebabToSnakeCase`.
- New `getCurrentUrl` helper for getting the current URL with query parameters.
- New `cleanUrlParams` helper for cleaning URL parameters.
- New project info helpers: `getPluginVersion`, `getPluginName`, `getPluginTextDomain`, `getThemeVersion`, `getThemeName`, `getThemeTextDomain`, `getPluginDetails`.
- New constants for API handing in the `src/Rest/Routes/AbstractRoute.php`.

### Fixed
- Typo in block enqueue method for `getBlockFrontentScriptHandle` to `getBlockFrontendScriptHandle` method.

### Removed
- `src/Blocks/AbstractBlocks.php` method `renderWrapperView` is removed.
- `Config/AbstractConfigData.php` is removed as it is no longer needed. Use helpers instead.
- `Config/ConfigDataInterface.php` is removed as it is no longer needed.
- `LabelGeneratorTrait` is removed as it is no longer needed.
- All Enqueue methods no longer use `manifest` interface.
- `Exception/ComponentException.php` is removed as it is no longer needed.
- `Exception/FailedToLoadView.php` is removed as it is no longer needed.
- `Exception/FileMissing.php` is removed as it is no longer needed.
- `src/Exception/InvalidCallback.php` is removed as it is no longer needed.
- `src/Exception/InvalidNouns.php` is removed as it is no longer needed.
- `src/Exception/InvalidService.php` is removed as it is no longer needed.
- `src/Exception/PluginActivationFailure.php` is removed as it is no longer needed.
- `Exception/InvalidBlock.php` methods: `missingBlocksException`, `missingComponentsException`, `missingNameException`, `missingComponentNameException`, `missingVariationNameException`, `missingViewException`, `missingRenderViewException`, `missingSettingsManifestException`, `missingWrapperManifestException`, `missingComponentManifestException`, `missingWrapperViewException`, `missingNamespaceException`, `missingSettingsKeyException`, `wrongFunctionUsedException`, `missingFileException`, is removed as it is no longer needed.
- `src/Exception/InvalidManifest.php` method: `missingManifestItemException`.
- `src/Exception/InvalidPath.php` method: `fromUri`.
- `outputCssVariablesGlobal` method no longer requires parameters.
- `outputCssVariables` method is not longer supporting `globalManifest` parameter.
- all methods from `src/Helpers/ErrorLoggerTrait.php` trait are removed.
- All manifest classes are removed, `src/Manifest/AbstractManifest.php`, `src/Manifest/ManifestExample.php`, `src/Manifest/ManifestInterface.php`.

### Deprecated
- `Helpers>Components` class is deprecated and will be removed in the next major release. Use `Helpers>Blocks` class instead.
- `Helpers>Components>renderPartial` method is deprecated and will be removed in the next major release. Use `Helpers>Helpers>render` method instead.
- `Helpers>Components>getManifest` method is deprecated and will be removed in the next major release. Use `Helpers>Helpers>getManifestByDir` method instead.

### CODE REVIEW

**Review by DjidjaleskaSandra — DISMISSED** [review]

File: src/Misc/VersionCli.php [hunk 1]
```diff
@@ -90,32 +90,34 @@ public function getDoc(): array
...
+
 		$this->getIntroText($assocArgs);
 
 		// Get Props.
 		$version = $this->getArg($assocArgs, 'version');
 
-		$assocArgs['skip_existing'] = 'true';
+		$assocArgs[AbstractCli::ARG_SKIP_EXISTING] = 'true';
 		// translators: %s is replaced with the new version number.
 		$assocArgs['actionOutput'] = \sprintf(\__("version number changed to %s.", 'eightshift-libs'), $version);
```
Comment (piqusy) [inline src/Misc/VersionCli.php:102]: Placeholder textdomain?

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

Left a few suggestions, mostly the same... three of a kind 😄

File: src/Init/InitAllCli.php [hunk 2]
```diff
@@ -85,46 +64,38 @@ public function getDoc(): array
...
+				)
+			);
 		}
 
-		if (!$groupOutput) {
-			$this->cliLog('We have moved everything we have to your project. Please type `npm start` in your terminal to kickstart your assets bundle process.', "M");
-			$this->cliLog('Happy developing!', "M");
+		if (!$assocArgs[self::ARG_GROUP_OUTPUT]) {
+			$this->cliLogAlert(
+				'All the files have been created, you can start working on your awesome project!',
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Init/InitAllCli.php:94]: no need as this are cli commands and they are not affected by the translations

File: src/Init/InitAllCli.php [hunk 3]
```diff
@@ -85,46 +64,38 @@ public function getDoc(): array
...
+				'success',
+				\__('Ready to go!', 'eightshift-libs')
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Init/InitAllCli.php:96]: fixed

File: src/Init/theme/package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	"license": "MIT",
+	"scripts": {
+		"lintStyle": "stylelint **/*.scss",
+		"lintJs": "eslint src/ assets/",
+		"lint": "npm run lintJs && npm run lintStyle && composer standards:check -- --filter=gitstaged",
+		"start": "webpack --watch --mode development",
+		"build": "webpack --progress --mode production"
+	},
+	"devDependencies": {
+		"husky": "^8.0.3",
```
[the remark this answers is not in the document]
Comment (iruzevic) [reply src/Init/theme/package.json:14]: in the next pr

**Review by piqusy — APPROVED** [review]
Looking good 🚀

**Pull request merged**